### ISSUE TITLE
TRAMP issue in Emacs 26

### ISSUE THREAD

**Issue by Alexander-Shukaev** [magit/with-editor#29]
In Emacs 26, attempting to commit under TRAMP (e.g. via `sudo` method) results in

```
Debugger entered--Lisp error: (wrong-number-of-arguments (6 . 7) 5)
  tramp-make-tramp-file-name("sudo" #("root" 0 4 (tramp-default t)) "host" "/.git/COMMIT_EDITMSG" nil)
  with-editor-output-filter("WITH-EDITOR: 3484 OPEN /.git/COMMIT_EDITMSG\n")
  with-editor-process-filter(#<process git<1>> "WITH-EDITOR: 3484 OPEN /.git/COMMIT_EDITMSG\n" t)
  (lambda (proc str) (magit-process-filter proc str) (with-editor-process-filter proc str t))(#<process git<1>> "WITH-EDITOR: 3484 OPEN /.git/COMMIT_EDITMSG\n")
```

because the signature of `tramp-make-tramp-file-name` now accepts `port` at the previous position of `localname`:

```
(defun tramp-make-tramp-file-name
    (method user domain host port localname &optional hop)
  "Constructs a Tramp file name from METHOD, USER, HOST and LOCALNAME.
When not nil, optional DOMAIN, PORT and HOP are used."
  ...)
```

Please, adapt.

**Comment by kyleam** [comment]
Testing #30 this with sudo, Emacs 26 seems very sluggish compared to
Emacs 25.  @Alexander-Shukaev, I'm curious if you see the same.

**Comment by Alexander-Shukaev** [comment]
Emacs 26 looks at least as fast as 25 and in some cases even faster for me.  By the way, see the recent issue I've discovered on Emacs 24 and 25 series along with the solution [[Garbage collection: Memory leak?](https://emacs.stackexchange.com/q/33418)].  Never ever going back to 25 or 24.

**Comment by kyleam** [comment]
> Emacs 26 looks at least as fast as 25 and in some cases even faster for me

Eh, sorry, I worded my comment poorly.  I wasn't talking about the
speed of Emacs 26 in general, just the tramp processing for the
with-editor calls.

After pressing `c c` in Emacs 26, it takes several seconds until I'm
thrown into the COMMIT_EDITMSG buffer.  Clearly something's off
(perhaps just with my install), but I haven't had time to look into it
more.  Anyway, even if others see the same, it's not directly related
to this issue or the fix in #30.
